### PR TITLE
Eliminate CAMLprim/CAMLexport in socketaddr.h

### PR DESCRIPTION
**Pull request by dra27**
This PR starts from removing the incorrect use of `CAMLexport` in `otherlibs/unix/socketaddr.h` and, er, sort of equivalently `CAMLprim` on two of the "same" declarations and `CAMLexport` on the third in `otherlibs/win32unix/socketaddr.h`.

The fact the declarations were different suggests - as with a lot of win32unix and unix - that merging them would be a good idea.

Then one spots that win32unix's version still has the `extern union sock_addr_union sock_addr;` global, the usage of which was eliminated 20 years ago in 803d9504aa89fe5756086448f1c09794b1c5d254 for the unix version and then removed completely 4 years later in https://github.com/ocaml/ocaml/commit/641657c87b1285b2e428469540c9506e37c04eb6! The consequential changes in C files in win32unix come from eliminating this global on the win32unix side and some obvious differences while comparing the file against its counterpart in `unix/`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -160,6 +160,7 @@ _build
 /otherlibs/win32unix/putenv.c
 /otherlibs/win32unix/rmdir.c
 /otherlibs/win32unix/socketaddr.c
+/otherlibs/win32unix/socketaddr.h
```
Comment (xavierleroy) [inline .gitignore]: Unnecessary?

File: otherlibs/win32unix/Makefile [hunk 2]
```diff
@@ -34,7 +34,7 @@ UNIX_FILES = access.c addrofstr.c chdir.c chmod.c cst2constr.c \
   exit.c getaddrinfo.c getcwd.c gethost.c gethostname.c \
   getnameinfo.c getproto.c \
   getserv.c gmtime.c mmap_ba.c putenv.c rmdir.c \
-  socketaddr.c strofaddr.c time.c unlink.c fsync.c
+  socketaddr.c strofaddr.c time.c unlink.c fsync.c socketaddr.h
```
Comment (xavierleroy) [inline otherlibs/win32unix/Makefile]: The C stub files in win32unix are compiled with `-I../unix`, so do we really need to copy `socketaddr.h` from `../unix`?

**Review by xavierleroy — COMMENTED** [review]
Looks good to me overall, but something is not clear to me, see below.

File: otherlibs/win32unix/Makefile (hunk 2 above)
Comment (dra27) [reply otherlibs/win32unix/Makefile]: I hadn't registered that, but as it happens I think it is still necessary - it's included in `HEADERS` and so installed and the general rule in `Makefile.otherlibs.common` doesn't know differently.

Comment (xavierleroy) [reply otherlibs/win32unix/Makefile]: OK, that's one reason for copying the header.

**Review by xavierleroy — APPROVED** [review]

**Review by xavierleroy — CHANGES_REQUESTED** [review]
The Windows build is broken.  One way or another, `socketaddr.h` is passed to `ocamlmklib`.

**Comment by dra27** [comment]
Whoops - hopefully the revised version works as originally intended!

**Comment by xavierleroy** [comment]
Now it's the installation of socketaddr.h that fails...

**Comment by dra27** [comment]
Indeed (I think it's also specific to the case where C dependencies are _not_ generated). A solution is to have the install target depend on `$(HEADERS)` but this is getting silly. `$(HEADERS)` is _only_ used for the `install` target, so I take your original point that we have `-I../unix` already and simply change `HEADERS` to have `../socketaddr.h`... hopefully CI finally passes! I removed the entry from `.gitignore` therefore as well.

**Review by xavierleroy — APPROVED** [review]
CI is finally happy, and I am too!

**Pull request merged**